src/Build/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -780,14 +792,24 @@ public string[] InputResultsCacheFiles
 
         /// <summary>
         /// Output cache file where MSBuild will write the contents of its build result caches during EndBuild.
-        /// Setting this also turns on isolated builds.
+        /// If the isolation mode is set to <see cref="ProjectIsolationMode.False"/>,
+        /// this sets the isolation mode to <see cref="ProjectIsolationMode.True"/>.
         /// </summary>
         public string OutputResultsCacheFile
         {
             get => _outputResultsCacheFile;
             set => _outputResultsCacheFile = value;
         }
 
+        /// <summary>
+        /// The set of specified targets to build.
+        /// </summary>
+        public HashSet<string> Targets
```

Comment:
This is adding to the public API and I'm not sure it's entirely clear how this is supposed to interact with the set of Targets provided when building.

Reply:
In particular I'm referring to the targets provided on `BuildRequestData`

Reply:
They're not supposed to interact with the targets in `BuildRequestData`. I placed them into `BuildParameters` because I considered them a build parameter used with the `OutputResultsCacheFile` build parameter. I don't think it should be added to the public API though... Is there an attribute I can use to ensure it's not included in it? Otherwise, I'll just have to pass it as an argument to `EndBuild` from `XMake.cs`.

Reply:
I definitely agree that this feels wrong. I'm also not sure I understand the goal of passing this in here. Is the list of targets to put into the cache file the list of targets passed to the entry-point (what you'd see in xmake.cs) or the list computed by the graph for a given node? I expected the latter.

Reply:
It should be the latter. I've fixed it by obtaining the targets from the configuration's `BuildRequestConfiguration.TargetNames`, which are set through `BuildRequestData`.